examples/seq2seq/finetune.py

```diff
@@ -137,10 +142,16 @@ def _step(self, batch: dict) -> Tuple:
         outputs = self(src_ids, attention_mask=src_mask, decoder_input_ids=decoder_input_ids, use_cache=False)
         lm_logits = outputs[0]
         if self.hparams.label_smoothing == 0:
+            model_type = self.config.model_type
+            if model_type == "fsmt":
+                vocab_size = self.model.config.tgt_vocab_size
```

Comment:
you can do 
`self.tgt_vocab_size = self.config.tgt_vocab_size if model_type == 'fsmt` else self.config.vocab_size` in init and only evaluate it once.

Reply:
Indeed! Done. (this could lead to a potential problem elsewhere if suddenly it's `src_vocab_size` is needed but it won't be the case with eval).

I also moved setting `self.model_type` into init - for the same reason. let me know if that wasn't a good move.